tests/functional/conftest.py

```diff
@@ -0,0 +1,37 @@
+import subprocess
+
+import pytest
+from tenacity import retry, stop_after_delay, wait_fixed
+
+
+@retry(wait=wait_fixed(5), stop=stop_after_delay(20))
+def check_dcgm_exporter_service():
+    dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+    result = subprocess.run(
+        f"sudo snap services {dcgm_exporter_service}".split(),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    assert " active" in result.stdout.strip(), f"{dcgm_exporter_service} service is not active"
```

Comment:
wouldn't `systemctl --is-active --quiet` be simpler here? The output of snap services seem very human-centric to me